assets/js/blocks/single-product/edit/index.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Placeholder, Button, PanelBody } from '@wordpress/components';
+import { withProduct } from '@woocommerce/block-hocs';
+import BlockErrorBoundary from '@woocommerce/base-components/block-error-boundary';
+import EditProductLink from '@woocommerce/editor-components/edit-product-link';
+import { singleProductBlockPreview } from '@woocommerce/resource-previews';
+import { InspectorControls, useBlockProps } from '@wordpress/block-editor';
+import { ProductResponseItem } from '@woocommerce/types';
+import { ErrorObject } from '@woocommerce/editor-components/error-placeholder';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import ApiError from './api-error';
+import SharedProductControl from './shared-product-control';
+import EditorBlockControls from './editor-block-controls';
+import LayoutEditor from './layout-editor';
+import { BLOCK_ICON } from '../constants';
+import metadata from '../block.json';
+import { Attributes } from '../types';
+
+interface EditorProps {
+	className: string;
+	attributes: {
+		productId: number;
+		isPreview: boolean;
+	};
+	setAttributes: ( attributes: Attributes ) => void;
+	error: string | ErrorObject;
+	getProduct: () => void;
+	product: ProductResponseItem;
+	isLoading: boolean;
+	clientId: string;
+}
+
+/**
+ * Component to handle edit mode of the "Single Product Block".
+ *
+ * @param {Object}            props               Incoming props for the component.
+ * @param {string}            props.className
+ * @param {Object}            props.attributes    Incoming block attributes.
+ * @param {function(any):any} props.setAttributes Setter for block attributes.
+ * @param {string}            props.error
+ * @param {function(any):any} props.getProduct
+ * @param {Object}            props.product
+ * @param {boolean}           props.isLoading
+ * @param {string}            props.clientId
```

Comment:
Not a blocker either, but given that this component is typed with TS, could we skip the JSDoc types? Or is it needed for something else?

Reply:
You're correct. These files inside the `edit` folder was copied from the old Single Product block and they were JS files so we missed these annotations in here, thank you for reporting that